.github/workflows/ci.yml

```diff
@@ -77,47 +77,42 @@ jobs:
               echo 'Not going to push build'
           fi
   windows-mingw:
-    name: Windows (win32) using mingw
-    runs-on: ubuntu-latest
-    needs: [check-code-formatting]
-    container:
-      image: openrct2/openrct2-build:0.3.1-mingw
-    steps:
-      - name: Checkout
-        uses: actions/checkout@v2
-      - name: ccache
-        uses: hendrikmuhs/ccache-action@v1
-        with:
-          key: windows-mingw
-      - name: Build OpenRCT2
-        run: |
-          sudo su
-          mkdir bin && cd bin
-          cmake .. -G Ninja -DCMAKE_TOOLCHAIN_FILE=../CMakeLists_mingw.txt -DCMAKE_BUILD_TYPE=MinSizeRel -DDISABLE_IPO=on -DFORCE32=on -DBUILD_SHARED_LIBS=ON -DENABLE_SCRIPTING=OFF
-          ninja -k0
-  windows-mingw-nt51:
-    name: Windows (win32, WinNT5.1) using mingw
+    name: Windows (${{ matrix.platform_name }}) using mingw
     runs-on: ubuntu-latest
     needs: check-code-formatting
     container:
       image: openrct2/openrct2-build:0.3.1-mingw
+    strategy:
+      fail-fast: false
+      matrix:
+        platform: [win32, NT5.1]
+        include:
+          - platform: win32
+            platform_name: win32
+            cache_key: windows-mingw
+            build_flags: -DBUILD_SHARED_LIBS=ON -DENABLE_SCRIPTING=OFF
+          - platform: NT5.1
+            platform_name: win32, NT5.1
+            cache_key: windows-mingw-nt51
+            build_flags: -DDISABLE_HTTP=Off -DENABLE_SCRIPTING=ON -DCMAKE_CXX_FLAGS="-Wno-error=cast-function-type -Wno-error=unused-function" -DSTATIC=on -DMINGW_TARGET_NT5_1=ON
     steps:
       - name: Checkout
         uses: actions/checkout@v2
       - name: ccache
         uses: hendrikmuhs/ccache-action@v1
         with:
-          key: windows-mingw-nt51
+          key: ${{ matrix.cache_key }}
       - name: Build OpenRCT2
         run: |
           sudo su
           mkdir bin && cd bin
-          cmake .. -G Ninja -DCMAKE_TOOLCHAIN_FILE=../CMakeLists_mingw.txt -DCMAKE_BUILD_TYPE=MinSizeRel -DDISABLE_IPO=on -DDISABLE_HTTP=Off -DFORCE32=on -DENABLE_SCRIPTING=ON -DCMAKE_CXX_FLAGS="-Wno-error=cast-function-type -Wno-error=unused-function" -DSTATIC=on -DMINGW_TARGET_NT5_1=ON
+          cmake .. -G Ninja -DCMAKE_TOOLCHAIN_FILE=../CMakeLists_mingw.txt -DCMAKE_BUILD_TYPE=Release -DDISABLE_IPO=on -DFORCE32=on ${{ matrix.build_flags }}
```

Comment:
Why the change from `MinSizeRel` to `Release`? (this is on commit https://github.com/OpenRCT2/OpenRCT2/pull/14479/commits/b513c0f297bdd394d65cdc728bda7a29128f2ca4)

Reply:
I thought `MinSizeRel` optimized for size instead of speed.

Reply:
From what I searched it seems to try to do both, but people do recommend `Release` indeed, so I'm ok with this.